src/example_report.rs

```diff
@@ -0,0 +1,90 @@
+use std::convert::From;
+
+#[cfg(feature = "use_expectest")]
+use expectest::core::TestResult as ExpectestResult;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Failure {
+    pub message: Option<String>,
+}
+
+impl Failure {
+    pub fn new<S>(message: Option<S>) -> Self
+        where String: From<S>
```

Comment:
It is used for `Failure` creation from either a `str` (literal) or a `String` (`format!(…)`).